app/services/file_version_membership_diff.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+# @abstract Returns a hash with the differences between the file memberships of two WorkVersions. Renamed files
+# is array of tuples, each with a FileVersionMembership, representating the two changed memberships. Added and deleted
+# files are returned as arrays of FileVersionMembership objects.
+# @example Given two objects with two different titles
+#   >  FileVersionMembershipDiff.call(obj1, obj2)
+#   => {
+#        renamed:
+#          [
+#            [FileVersionMembership, FileVersionMembership],
+#            [FileVersionMembership, FileVersionMembership]
+#          ]
+#        added: [FileVersionMembership, FileVersionMembership],
+#        deleted: [FileVersionMembership]
+#      }
+#
+class FileVersionMembershipDiff
+  # @param obj1 [WorkVersion]
+  # @param obj2 [WorkVersion]
+  def self.call(*args)
+    new(*args).diff
+  end
+
+  attr_reader :base_version, :comparison_version
+
+  def initialize(*args)
+    @base_version = args[0]
+    @comparison_version = args[1]
+  end
+
+  def diff
+    { renamed: renamed, deleted: deleted, added: added }
+  end
+
+  # @return [Array<[FileVersionMembership, FileVersionMembership]>]
+  def renamed
+    renamed_file_resource_ids.map do |id|
+      original_file = base_version_memberships.where(file_resource_id: id).first
+      renamed_file = comparison_version_memberships.where(file_resource_id: id).first
```

Comment:
The above two lines query the database unnecessarily with the `where`. We can change this to `Enumerable#find` (not to be confused with the ActiveRecord method of the same name) to do this finding in memory rather than another round trip to the DB